packages/bigquery/src/job.js

```diff
@@ -344,6 +344,27 @@ Job.prototype.listenForEvents_ = function() {
 };
 
 /**
+ * Convenience method that wraps the `complete` and `error` events in a
+ * Promise.
+ *
+ * @return {promise}
+ *
+ * @example
+ * job.promise().then(function(metadata) {
+ *   // The job is complete.
+ * }, function(err) {
+ *   // An error occurred during the job.
+ * });
+ */
+Job.prototype.promise = function() {
```

Comment:
Since this is a public method, please define this earlier in the file to alpha-sort with the public methods.